### DISCUSSION TITLE
Type field when creating an item with update

### DISCUSSION THREAD

**Discussion by pauleikom** [sensedeep/dynamodb-onetable#131] [Q&A] [answered]
When update creates an item that doesn't already exist the type field isn't set on the created item. Is there a way to set the type field in this situation? 

```await User.update(user, { exists: null });```

**Comment by mobsense** [comment]
I'll look into this more. I think the issue is that if we universally add the type, then every update is redundantly updating the type.

**Comment by mobsense** [comment, marked as answer]
Looks doable. If exists == null or false, then we can update the type. For the normal case of exists === undefined or if exists === true, then we won't update type. ;-)

**Reply by pauleikom** [reply]
that would be great! thank you
